packages/photon/src/index.ts

```diff
@@ -26,11 +45,11 @@ const PHOTON_BASE_URL = 'https://i0.wp.com';
  *
  * Photon documentation: http://developer.wordpress.com/docs/photon/
  *
- * @param {string} imageUrl - the URL of the image to run through Photon
- * @param {object} [opts] - optional options object with Photon options
- * @returns {string} The generated Photon URL string
+ * @param imageUrl - the URL of the image to run through Photon
+ * @param [opts] - optional options object with Photon options
+ * @returns The generated Photon URL string
  */
-export default function photon( imageUrl, opts ) {
+export default function photon( imageUrl: string, opts?: PhotonOpts ): string | null {
```

Comment:
We could change the return type to just `string` and if the `imageUrl` fails to parse, just return it as it is.

All `photon()` usages in the codebase assume that the function returns an URL string and never check for success.

Reply:
That makes sense to me. I wasn't sure if it was used outside of Calypso, in which case this would be a breaking change.

Reply:
Actually, it looks like `[safeImageUrl()](https://github.com/Automattic/wp-calypso/blob/31d3844d216c261ed9bff5ea0abadc0aa0d78b4b/client/lib/safe-image-url/index.js#L45)` uses `photon()` under the hood and relies on it returning null if it's not valid. Other places in the codebase rely on `safeImageUrl` returning null if the image won't work: https://github.com/Automattic/wp-calypso/blob/31d3844d216c261ed9bff5ea0abadc0aa0d78b4b/client/lib/post-normalizer/rule-keep-valid-images.js#L6

Reply:
OK, let's keep the `string | null` return type then.